sql/core/src/main/scala/org/apache/spark/sql/execution/r/MapPartitionsRWrapper.scala

```diff
@@ -25,6 +25,21 @@ import org.apache.spark.sql.Row
 import org.apache.spark.sql.types.{BinaryType, StructField, StructType}
 
 /**
+ * A function wrapper that applies the given R function to each partition of each group.
+ */
+private[sql] case class MapGroupsRWrapper(
```

Comment:
Thank you for the explanation, @sun-rui !

As I understand from previous notes:
We want to pass multiple groups to a R worker and call the R function on each group separately, right ?

I'm trying to understand what is the major difference between this approach and repartition .

Because conceptually repartition was also passing multiple groups to a R worker (guaranteed that each group will be in one partition, but there can be multiple groups in a partition).   

I'm fine with this optimization, just trying to compare.